drivers/fpga/fpga_ice40.c

```diff
@@ -262,7 +262,11 @@ static int fpga_ice40_load_gpio(const struct device *dev, uint32_t *image_ptr, u
 	LOG_DBG("Delay %u us", config->creset_delay_us);
 	fpga_ice40_delay(2 * config->mhz_delay_count * config->creset_delay_us);
 
-	__ASSERT(gpio_pin_get_dt(&config->cdone) == 0, "CDONE was not high");
+	if (gpio_pin_get_dt(&config->cdone) != 0) {
```

Comment:
`!=` can be omitted.

(a small suggestion.)

Reply:
Yes, it can. But I'm always trying to avoid automatic conversions from integers to booleans, this doesn't improve readability in my opinion.

Reply:
> != can be omitted.

That is a direct violation of the [Zephyr Coding Guidelines Rule 85, which is based on MISRA C Rule 14.4](https://docs.zephyrproject.org/latest/contribute/coding_guidelines/index.html).

Reply:
> That is a direct violation of the [Zephyr Coding Guidelines Rule 85, which is based on MISRA C Rule 14.4](https://docs.zephyrproject.org/latest/contribute/coding_guidelines/index.html).

This makes it even easier and overrules the taste-preference.

Reply:
Good to know! I could never decide myself between `if (err) { ... }` or `if (ret < 0) { ... }` :)